framework/validators/CBooleanValidator.php

```diff
@@ -66,6 +66,25 @@ protected function validateAttribute($object,$attribute)
 			));
 		}
 	}
+	
+	/**
+	 * Validates a static value to see if it is a valid boolean.
+	 * This method is provided so that you can call it directly without going
+	 * through the model validation rule mechanism.
+	 * @param mixed $value the value to be validated
+	 * @return boolean whether the value is a valid boolean
+	 * @since 1.1.17
+	 */
+	public function validateValue($value)
+	{
+		if($this->allowEmpty && $this->isEmpty($value))
```

Comment:
@cebe 'validateAttribute' now uses 'validateValue' method. However, why shouldn't this check be done here? The same check is used in 'validateValue' method in CEmailValidator class.